WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -17,19 +17,74 @@ private struct QueryRef<Query> {
 /// Subclasses need to implement queriesChanged() and look at activeQueries and
 /// their internal state to decide if they need to act.
 ///
+/// If your `State` type confrorms to the `CachableState` protocol, the `QueryStore`
+/// will transparently persist/load cached State and purge in-memory State when there are no
+/// longer any active `Queries` associated with the `Store`.
+////
+/// You can also manually force writing of the cache file by calling `storeCachedState()`.
+///
+
 open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
+    private let initialState: State
+
     fileprivate var activeQueryReferences = [QueryRef<Query>]() {
         didSet {
+            if activeQueryReferences.isEmpty, let encodableState = state as? Encodable, let url = try? type(of: self).cacheFileURL() {
+                // If we don't have any active queries, and the `state` conforms to `Encodable`, let's use this as our cue to persist the data
+                // to disk and get rid of the in-memory cache.
+                inMemoryState = nil
+                encodableState.storeCachedState(at: url)
+            }
+
             queriesChanged()
         }
     }
 
+    /// In-memory storage for `state`.
+    private var inMemoryState: State?
+
+    /// Facade for the `state`.
+    ///
+    /// It allows for the lazy-loading of `state` from cache, when `State` conforms to `Codable`.
+    override public final var state: State {
+        get {
+            // If the in-memory State is populated, just return that.
+            guard inMemoryState == nil else {
+                return inMemoryState!
```

Comment:
Minor nitpick, but I always freak out a bit when I see a `!` 😁 Can we turn the `guard` into an `if let`?